src/main/java/com/noticemedan/map/model/kdtree/Forest.java

```diff
@@ -1,16 +1,71 @@
 package com.noticemedan.map.model.kdtree;
 
-import com.noticemedan.map.model.OsmElement;
+import com.noticemedan.map.data.BinaryMapData;
+import com.noticemedan.map.data.OsmMapData;
 import com.noticemedan.map.model.OsmElement;
 import com.noticemedan.map.model.utilities.Rect;
-import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.util.*;
+
 
-import java.util.ArrayList;
-import java.util.List;
 
-@AllArgsConstructor
-public class Forest implements ForestInterface {
+public class Forest implements ForestInterface{
 	private KDTree trees[];
+	private final OsmMapData osmMapData = new OsmMapData();
+	private final String binaryID = UUID.randomUUID().toString();
+	@Getter
+	private List<OsmElement> coastlines = this.osmMapData.getOsmCoastlineElements().toJavaList();
+
+	public Forest() {
+		int maxNumberOfElementsAtLeaf = 100;
+		List<OsmElement> osmMaterialElements = this.osmMapData.getOsmElements().toJavaList();
+		OsmElement[][] osmMaterialElementArray = new OsmElement[3][];
+
+		List<OsmElement> zoom0 = new LinkedList<>();
+		List<OsmElement> zoom1 = new LinkedList<>();
+		List<OsmElement> zoom2 = new LinkedList<>();
+
+		osmMaterialElements.forEach(m -> {
```

Comment:
Hvad står m for?

Reply:
osmMaterial åbenbart, har ikke lavet det

Reply:
refactor det til `osmElement`, så ser det bedre ud:)